### PR TITLE
Save new goal

### PR DESCRIPTION
**Pull request by emilyb7**
- Refactors the avatars component to be a controlled component
- Selected avatar is shown using blue border
- new goal is added to goals array in app state
- user redirected back to goals list
- goal shown in list

### CODE REVIEW

File: app/actions/add-new-goal.js [hunk 1]
```diff
@@ -1,10 +1,30 @@
-import { INPUT_GOAL } from './../action_types.js';
+import * as types from './../action_types.js';
```
Comment (des-des) [inline app/actions/add-new-goal.js]: I think you save characters here with
```js
import { INPUT_GOAL } from '../action_types.js' 
```

It is in the case when you will just pass on the object the import * as gives you without trying to get its properties.

Comment (des-des) [inline app/actions/add-new-goal.js]: Does this make sense?

This is just a stylistic point so I am not requesting changes

File: app/containers/main-container.jsx [hunk 2]
```diff
@@ -50,7 +55,9 @@ const mapStateToProps = state => ({
 
 const mapDispatchToProps = dispatch => ({
   stepAddGoal: () => { dispatch(action_stepAddGoal()) },
-  onInputGoal: (text) => { dispatch(action_inputGoal(text)) },
+  onInputGoal: (text) => { dispatch(actionsAddNewGoal.inputGoal(text)) },
```
Comment (des-des) [inline app/containers/main-container.jsx]: No!

You can just pass actionsAddNewGoal as the second argument to the connect function (redux will sort the rest!)

File: app/actions/add-new-goal.js (hunk 1 above)
Comment (emilyb7) [inline app/actions/add-new-goal.js]: @des-des i'm sure you're making a good point, but that literally makes no sense 😉

Comment (emilyb7) [inline app/actions/add-new-goal.js]: as in, your sentence makes no sense so i think your point got a bit lost

Comment (emilyb7) [inline app/actions/add-new-goal.js]: ok i think i get what you mean now, that it's best to destructure the object...

File: app/containers/main-container.jsx (hunk 2 above)
Comment (emilyb7) [inline app/containers/main-container.jsx]: @des-des can you explain a bit more please? not sure what i'm supposed to be doing...

File: app/actions/add-new-goal.js (hunk 1 above)
Comment (des-des) [inline app/actions/add-new-goal.js]: Yes!

File: app/containers/main-container.jsx (hunk 2 above)
Comment (des-des) [inline app/containers/main-container.jsx]: https://github.com/CYPIAPT-LNDSE/goals-app/pull/130

**Pull request merged**